core/templates/services/html-length.service.ts

```diff
@@ -126,19 +147,23 @@ export class HtmlLengthService {
       nonTextNode: string, calculationType: CalculationType): number {
     if (nonTextNode.includes('oppia-noninteractive-math')) {
       return 1;
-    }
-    const altTextMatch = nonTextNode
-      .match(/alt-with-value="&amp;quot;([^&]*)&amp;quot;"/);
-
-    let text = [];
-    if (altTextMatch && altTextMatch[1]) {
-      const altText = altTextMatch[1];
-      if (calculationType === CALCULATION_TYPE_WORD) {
-        text = altText.trim().split(' ');
-      } else if (calculationType === CALCULATION_TYPE_CHARACTER) {
-        text = altText.trim().split(' ').join('').split('');
+    } else if (nonTextNode.includes(
+      'oppia-noninteractive-collapsible') || nonTextNode.includes(
```

Comment:
Given that you are passing a single non-text node, can you do something more deterministic to get the type of that node (or pass it in explicitly)?

In general, use string inclusion as little as possible when dealing with HTML. It's better to semantically extract the values of specific attributes and then you can parse them as needed. But don't treat this complex object as just "a string".

Reply:
Parse the string, retrieve the ID, and use that ID to determine which calculation approach to use.

Reply:
Done!